test/Interop/Cxx/templates/Inputs/ForwardDeclaredSpecialization.h

```diff
@@ -0,0 +1,63 @@
+#ifndef FORWARD_DECLARED_SPECIALIZATION_H
+#define FORWARD_DECLARED_SPECIALIZATION_H
+
+// Basic template definition
+template <typename T>
+struct BasicTemplate {
+  T value;
+};
+
+// Case 1: Forward-declared specialization (should NOT import)
+template <>
+struct BasicTemplate<int>;
+typedef BasicTemplate<int> ForwardDeclaredInt;
+
+// Case 2: Complete specialization (should import successfully)
+template <>
+struct BasicTemplate<double> {
+  double value;
+  double getValue() const { return value; }
+};
+typedef BasicTemplate<double> CompleteDouble;
+
+// Case 3: Specialization defined after typedef (should import)
+template <>
+struct BasicTemplate<float>;
+typedef BasicTemplate<float> FloatTypedef;
+
+template <>
+struct BasicTemplate<float> {
+  float value;
+};
+
+// Case 4: For comparison - forward-declared non-templated struct (have same behavior)
```

Comment:
nit: it’s not clear what the behavior is the same as. Better to make it explicit imo:

```suggestion
// Case 4: For comparison - forward-declared non-templated struct (also does not import)
```

This suggestion isn’t blocking, i.e., no need to change the comment and be forced to re-run CI.